File: crud_contact.py
# Struktur data menggunakan list untuk menyimpan kontak
contacts = []

# Fungsi untuk menambah kontak baru
def add_contact():
    name = input("Enter name: ")
    phone = input("Enter phone number: ")
    email = input("Enter email: ")
    contact = {'name': name, 'phone': phone, 'email': email}
    contacts.append(contact)
    print("Contact added successfully!")

# Fungsi untuk menampilkan semua kontak
def list_contacts():
    if len(contacts) == 0:
        print("No contacts found.")
    else:
        for i, contact in enumerate(contacts, start=1):
            print(f"{i}. {contact['name']} - {contact['phone']} - {contact['email']}")

# Fungsi untuk mengedit kontak
def edit_contact():
    list_contacts()
    try:
        index = int(input("Enter the number of the contact to edit: ")) - 1
        if 0 <= index < len(contacts):
            print(f"Editing contact: {contacts[index]['name']}")
            contacts[index]['name'] = input("Enter new name: ")
            contacts[index]['phone'] = input("Enter new phone number: ")
            contacts[index]['email'] = input("Enter new email: ")
            print("Contact updated successfully!")
        else:
            print("Invalid contact number.")
    except ValueError:
        print("Invalid input. Please enter a number.")

# Fungsi untuk menghapus kontak
def delete_contact():
    list_contacts()
    try:
        index = int(input("Enter the number of the contact to delete: ")) - 1
        if 0 <= index < len(contacts):
            contacts.pop(index)
            print("Contact deleted successfully!")
        else:
            print("Invalid contact number.")
    except ValueError:
        print("Invalid input. Please enter a number.")

# Fungsi untuk menampilkan menu
def show_menu():
    print("\nContact Management System")
    print("1. Add contact")
    print("2. List contacts")
    print("3. Edit contact")
    print("4. Delete contact")
    print("5. Exit")
    return input("Choose an option: ")

# Fungsi utama untuk menjalankan aplikasi
def main():
    while True:
        choice = show_menu()

        if choice == '1':
            add_contact()
        elif choice == '2':
            list_contacts()
        elif choice == '3':
            edit_contact()
        elif choice == '4':
            delete_contact()
        elif choice == '5':
            print("Exiting program...")
            break
        else:
            print("Invalid option, please try again.")

if __name__ == "__main__":
    main()

File: crudcontact.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

// Struktur untuk menyimpan informasi kontak
struct Contact {
    string name;
    string phone;
    string email;
};

// Vektor untuk menyimpan daftar kontak
vector<Contact> contacts;

// Fungsi untuk menambah kontak baru
void addContact() {
    Contact newContact;
    cout << "Enter name: ";
    cin.ignore(); // Mengabaikan newline sebelumnya
    getline(cin, newContact.name);
    cout << "Enter phone number: ";
    getline(cin, newContact.phone);
    cout << "Enter email: ";
    getline(cin, newContact.email);
    contacts.push_back(newContact);
    cout << "Contact added successfully!\n";
}

// Fungsi untuk menampilkan semua kontak
void listContacts() {
    if (contacts.empty()) {
        cout << "No contacts found.\n";
    } else {
        for (int i = 0; i < contacts.size(); ++i) {
            cout << i + 1 << ". " << contacts[i].name << " - " 
                << contacts[i].phone << " - " << contacts[i].email << endl;
        }
    }
}

// Fungsi untuk mengedit kontak
void editContact() {
    int index;
    cout << "Enter the number of the contact to edit: ";
    cin >> index;
    if (index > 0 && index <= contacts.size()) {
        cout << "Editing contact: " << contacts[index - 1].name << endl;
        cout << "Enter new name: ";
        cin.ignore();
        getline(cin, contacts[index - 1].name);
        cout << "Enter new phone number: ";
        getline(cin, contacts[index - 1].phone);
        cout << "Enter new email: ";
        getline(cin, contacts[index - 1].email);
        cout << "Contact updated successfully!\n";
    } else {
        cout << "Invalid contact number.\n";
    }
}

// Fungsi untuk menghapus kontak
void deleteContact() {
    int index;
    cout << "Enter the number of the contact to delete: ";
    cin >> index;
    if (index > 0 && index <= contacts.size()) {
        contacts.erase(contacts.begin() + index - 1);
        cout << "Contact deleted successfully!\n";
    } else {
        cout << "Invalid contact number.\n";
    }
}

// Fungsi untuk menampilkan menu
void showMenu() {
    cout << "\nContact Management System\n";
    cout << "1. Add contact\n";
    cout << "2. List contacts\n";
    cout << "3. Edit contact\n";
    cout << "4. Delete contact\n";
    cout << "5. Exit\n";
    cout << "Choose an option: ";
}

// Fungsi utama untuk menjalankan aplikasi
int main() {
    int choice;
    do {
        showMenu();
        cin >> choice;

        switch (choice) {
            case 1:
                addContact();
                break;
            case 2:
                listContacts();
                break;
            case 3:
                editContact();
                break;
            case 4:
                deleteContact();
                break;
            case 5:
                cout << "Exiting program...\n";
                break;
            default:
                cout << "Invalid option, please try again.\n";
        }
    } while (choice != 5);

    return 0;
}
